components/learn/modules/ROOT/pages/on-dapps.adoc

```diff
@@ -1,8 +1,160 @@
-= Dapps
+= Building a Dapp
 
-What is dapp vs a web 2.0 app?
-Components: CLI for managing contracts, connecting to the network, loading your contracts
-Introduce Network.js and hot-loader without sample project
-Starter kit for setting everything up
+Decentralized Applications (aka dapps or ÐApps) are web applications backed by Ethereum smart contracts. Instead of using a centralized server or database, these applications rely on the blockchain as a consensus and coordination layer. This leads to potentially unstoppable applications: anyone can spin up a copy of the frontend, and freely connect it to the public Ethereum network.
 
-See content from "writing a dapp" in SDK today
\ No newline at end of file
+In this guide, we will see how we can use OpenZeppelin libraries such as https://github.com/OpenZeppelin/openzeppelin-network.js[network-js] and the https://github.com/OpenZeppelin/solidity-loader[hot-loader] to build your dapp, as well as how the https://github.com/OpenZeppelin/starter-kit[Starter Kits] bind all pieces of a dapp together:
+
+  * <<network, Connecting to the Network>>
+  * <<hot-loader, Hot-loading your Contracts>>
+  * <<starter-kit, Kickstarting a Dapp using Starter Kits>>
+
+If you are in a rush, just skip over to <<starter-kit, the last section>>!
+
+[[network]]
+== Connecting to the Network
+
+When building a dapp, the first step is to connect to the Ethereum network. This usually involves the following steps:
+
+  . If the user is running Metamask, use the injected web3 provider to access the network, and monitor for network or account changes
+  . If not, fall back to a connection to a public Infura node
+
+To avoid having to code this logic on every dapp, we built https://github.com/OpenZeppelin/openzeppelin-network.js[`@openzeppelin/network`] to take care of it. It ships with a `useWeb3` https://reactjs.org/docs/hooks-intro.html[React hook] for getting a `web3Context` based on the logic above. This `web3Context` object provides access to a connected `web3` instance, network information, the user accounts, a function to https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1102.md[request account access], and more.
+
+The following snippet shows a small React app component that connects to the network using Metamask (falling back to Infura if not found), prompts the user for account access, and shows the user balance.
+
+```js
```

Comment:
highlight.js is supposed to support jsx, I'm not sure why this isn't properly rendered. @frangio ideas?

Reply:
We may need to add the XML language for the JSX highlighting. See this line: https://github.com/highlightjs/highlight.js/blob/master/src/languages/javascript.js#L203

Reply:
That did it, thanks!